### PR TITLE
Test cases

### PR DESCRIPTION
**Pull request by ntunin**
## Description
There was implemented test coverage for basic CollectionViewTools functionality

## Related tasks
[ISD-167](https://rosberry.atlassian.net/browse/ISD-167)

## Screenshots
<p align="center">

</p>

### CODE REVIEW

**Review by stason314 — APPROVED** [review]

File: CollectionViewToolsTests/CollectionViewManager/CollectionEdditingTests.swift [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        let title = "Title"
+        let reusableViewItem = TestReusableViewItem(title: title)
+        let kind = reusableViewItem.type.kind
+        // When
+        let cellItems = strings.map { string in
+            TestCollectionViewCellItem(text: string)
+        }
+        viewController.manager.sectionItems = [GeneralCollectionViewSectionItem(cellItems: cellItems,
+                                                                                reusableViewItems: [reusableViewItem])]
+        wait(for: [expectation], timeout: 10)
```
Comment (artemnovichkov) [inline CollectionViewToolsTests/CollectionViewManager/CollectionEdditingTests.swift:25]: I guess we should reduce timeouts

**Review by artemnovichkov — APPROVED** [review]

File: CollectionViewToolsTests/CollectionViewManager/CollectionViewManagerTests.swift [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        viewController.manager.scrollDelegate = delegate
+        //Then
+        wait(for: [expectation], timeout: 10)
+        viewController.manager.scroll(to: cellItems.last!, at: .top, animated: false)
+        XCTAssertGreaterThan(viewController.collectionView.contentOffset.y, viewController.view.bounds.height)
+        let shouldScroll = viewController.collectionView.delegate?.scrollViewShouldScrollToTop?(collectionView)
+        XCTAssertTrue(shouldScroll == true)
+        XCTAssertFalse(delegate.didScrollToTopTriggered)
+        viewController.manager.scroll(to: cellItems.first!, at: .top, animated: false)
+        XCTAssertEqual(viewController.collectionView.contentOffset.y, -44)
```
Comment (ReshetovE) [inline CollectionViewToolsTests/CollectionViewManager/CollectionViewManagerTests.swift]: This test is failed for me.

"Test Case '-[CollectionViewToolsTests.CollectionViewToolsTests testScroll]' started.
/Users/reshetove/Documents/iOS/Projects/Rosberry/Home/CollectionViewTools/CollectionViewToolsTests/CollectionViewManager/CollectionViewManagerTests.swift:116: error: -[CollectionViewToolsTests.CollectionViewToolsTests testScroll] : XCTAssertEqual failed: ("0.0") is not equal to ("-44.0")
Test Case '-[CollectionViewToolsTests.CollectionViewToolsTests testScroll]' failed (1.035 seconds)."

File: CollectionViewToolsTests/CollectionViewManager/CollectionEdditingTests.swift (hunk 1 above)
Comment (ntunin) [reply CollectionViewToolsTests/CollectionViewManager/CollectionEdditingTests.swift:25]: If the code works it should not matter

File: CollectionViewToolsTests/CollectionViewManager/CollectionViewManagerTests.swift (hunk 2 above)
Comment (ntunin) [reply CollectionViewToolsTests/CollectionViewManager/CollectionViewManagerTests.swift:116]: Could not reproduce

**Review by ReshetovE — APPROVED** [review]

File: CollectionViewToolsTests/CollectionViewManager/CollectionEdditingTests.swift (hunk 1 above)
Comment (antowkos) [reply CollectionViewToolsTests/CollectionViewManager/CollectionEdditingTests.swift:25]: I agree with @artemnovichkov, every redundant second of timeout increases total time of tests to be passed

**Pull request merged**